libshaderc/src/shaderc_cpp_test.cc

```diff
@@ -504,6 +504,51 @@ TEST_F(CppInterface, PreprocessingOnlyModeSecondOverridesDisassemblyMode) {
               HasSubstr("void main(){ }"));
 }
 
+TEST_F(CppInterface, SetIncluderCallbacks) {
+  const std::string kMinimalIncludeShader =
+      "void foo() {}\n"
+      "#include \"include/b\"\n";
+  // Create a fake file object, with specified fullpath and content.
+  struct FakeFile {
+    std::string fullpath = "Full/Path/Sample";
+    std::string content = "void main() {foo();}\n";
+  };
+  FakeFile fake_file;
+  shaderc_includer_callbacks callbacks{
+      // A response method that returns fake full path and file content as
+      // defined above.
+      [](void* user_data, const char* filename) {
```

Comment:
It is scattered around the spec, but http://stackoverflow.com/questions/17822131/resolving-ambiguous-overload-on-function-pointer-and-stdfunction-for-a-lambda has a good summary.